Data-Wrath.lua

```diff
@@ -181,128 +263,325 @@ function GladiusEx.Data.DefaultClassicon()
+    [48827] = 66, -- Avenger's Shield
+    [48825] = 65, -- Holy Shock
+    [53563] = 65, -- Beacon of Light
+    [35395] = 70, -- Crusader Strike
+    [66006] = 70, -- Divine Storm
+    [20066] = 70, -- Repentance
+    -- ROGUE
+    [48666] = 259, -- Mutilate
+    [14177] = 259, -- Cold Blood
+    [51690] = 260, -- Killing Spree
+    [13877] = 260, -- Blade Flurry
+    [13750] = 260, -- Adrenaline Rush
+    [36554] = 261, -- Shadowstep
+    [48660] = 261, -- Hemorrhage
+    [51713] = 261, -- Shadow Dance
+    -- PRIEST
+    [53007] = 256, -- Penance
+    [10060] = 256, -- Power Infusion
+    [33206] = 256, -- Pain Suppression
+    [34861] = 257, -- Circle of Healing
+    [15487] = 258, -- Silence
+    [48160] = 258, -- Vampiric Touch
+    -- DEATHKNIGHT
+    [55262] = 250, -- Heart Strike
+    [49203] = 251, -- Hungering Cold
+    [55268] = 251, -- Frost Strike
+    [51411] = 251, -- Howling Blast
+    [55271] = 252, -- Scourge Strike
+    -- MAGE
+    [44781] = 62, -- Arcane Barrage
+    [55360] = 63, -- Living Bomb
+    [42950] = 63, -- Dragon's Breath
+    [42945] = 63, -- Blast Wave
+    [44572] = 64, -- Deep Freeze
+    -- WARLOCK
+    [59164] = 265, -- Haunt
+    [47843] = 265, -- Unstable Affliction
+    [59672] = 266, -- Metamorphosis
+    [47193] = 266, -- Demonic Empowerment
+    [47996] = 266, -- Intercept Felguard
+    [59172] = 267, -- Chaos Bolt
+    [47847] = 267, -- Shadowfury
+    -- SHAMAN
+    [59159] = 262, -- Thunderstorm
+    [16166] = 262, -- Elemental Mastery
+    [51533] = 263, -- Feral Spirit
+    [30823] = 263, -- Shamanistic Rage
+    [17364] = 263, -- Stormstrike
+    [61301] = 264, -- Riptide
+    [51886] = 264, -- Cleanse Spirit
+    -- HUNTER
+    [19577] = 253, -- Intimidation
+    [34490] = 254, -- Silencing Shot
+    [53209] = 254, -- Chimera Shot
+    [60053] = 255, -- Explosive Shot
+    [49012] = 255, -- Wyvern Sting
+    -- DRUID
+    [53201] = 102, -- Starfall
+    [61384] = 102, -- Typhoon
+    [24858] = 102, -- Moonkin Form
+    [48566] = 103, -- Mangle (Cat)
+    [48564] = 103, -- Mangle (Bear)
+    [50334] = 103, -- Berserk
+    [18562] = 105, -- Swiftmend
+    [17116] = 105, -- Nature's Swiftness
+    [33891] = 105, -- Tree of Life
+    [53251] = 105, -- Wild Growth
+}
+
 function GladiusEx.Data.GetSpecializationInfoByID(id)
+    if specData[id] == nil then
+        return
+    end
+    return unpack(specData[id])
 end
 
 function GladiusEx.Data.GetNumSpecializationsForClassID(classID)
-  return 0
+    return 3
+end
+
+function GladiusEx.Data.GetSpecializationInfoForClassID(classID, specIndex)
+    local specID = classIDToSpecID[classID][specIndex]
+    local _, name, desc, icon, role, _, _ = GladiusEx.Data.GetSpecializationInfoByID(specID)
+    return specID, name, desc, icon, role, true, true -- isRecommended, isAllowed
+end
+
+function GladiusEx.Data.GetArenaOpponentSpec(id)
+    local unit = "arena" .. id
+
+    if GladiusEx.buttons[unit] and GladiusEx.buttons[unit].specID ~= nil then
```

Comment:
this is just `return GladiusEx.buttons[unit] and GladiusEx.buttons[unit].specID`

Reply:
Fixed.